python/ray/tune/stopper.py

```diff
@@ -81,33 +116,53 @@ def is_valid_function(cls, fn):
         return is_function
 
 
-class EarlyStopping(Stopper):
+class MaximumIterationStopper(Stopper):
+    """Stop trials after reaching a maximum number of iterations
+
+    Args:
+        max_iter (int): Number of iterations before stopping a trial.
+    """
+
+    def __init__(self, max_iter: int):
+        self._max_iter = max_iter
+        self._iter = defaultdict(lambda: 0)
+
+    def __call__(self, trial_id: str, result: Dict):
+        self._iter[trial_id] += 1
+        return self._iter[trial_id] >= self._max_iter
+
+    def stop_all(self):
+        return False
+
+
+class ExperimentPlateauStopper(Stopper):
+    """Early stop the experiment when a metric plateaued across trials.
+
+    Stops the entire experiment when the metric has plateaued
+    for more than the given amount of iterations specified in
+    the patience parameter.
+
+    Args:
+        metric (str): The metric to be monitored.
+        std (float): The minimal standard deviation after which
+            the tuning process has to stop.
```

Comment:
Instead, we should have something like an absolute threshold instead of std; std can have instabilities at small values anyways.

Reply:
But maybe it is out of scope :)

Reply:
I kept std as it was used in the `ExperimentPlateauStopper`. If we change this we should change this for both. Let's chat about this really quick offline later.